### PR TITLE
refactor(metrics): add 'greptime_' prefix for every metrics

### PR DESCRIPTION
**Pull request by zyy17**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

add `greptime_` prefix for every metrics

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [ ]  This PR does not require documentation updates.

### CODE REVIEW

**Review by killme2008 — COMMENTED** [review]
It's not a good idea to change these metrics name manualy, and we can add the prefix when exporting all metrics data by `prometheus::gather()` https://github.com/GreptimeTeam/greptimedb/blob/d973cf81f089d0ede8d6a6730443eff7a41d5e82/src/servers/src/metrics_handler.rs#L27

**Comment by zyy17** [comment]
> It's not a good idea to change these metrics name manualy, and we can add the prefix when exporting all metrics data by `prometheus::gather()`
> 
> https://github.com/GreptimeTeam/greptimedb/blob/d973cf81f089d0ede8d6a6730443eff7a41d5e82/src/servers/src/metrics_handler.rs#L27

Before this PR, I had already considered your advice and still think the plain metric name with a given prefix is the easiest way with the following advantages:

1. Only the greptimedb metrics with the prefix, other components with Prometheus metrics can still keep the original metrics(for example, opendal);
2. We rename those metrics once by text processing tools(for example, by `sed`). The new metrics should use `greptimedb_` prefix as the naming style in the code review;
3. The users can find out the metric definition easily by its exported name easily;

**Comment by killme2008** [comment]
I am not sure if this is necessary, because we can always specify a database to write metrics from GreptimeDB that achieves the same isolation. Another choice is adding a special tag (for example, the cluster name or id) to all the metrics.

What do you think? @sunng87 @fengjiachun

File: src/frontend/src/metrics.rs [hunk 1]
```diff
@@ -17,34 +17,34 @@ use prometheus::*;
 
 lazy_static! {
     pub static ref METRIC_HANDLE_SQL_ELAPSED: Histogram =
-        register_histogram!("frontend_handle_sql_elapsed", "frontend handle sql elapsed").unwrap();
+        register_histogram!("greptimedb_frontend_handle_sql_elapsed", "frontend handle sql elapsed").unwrap();
     pub static ref METRIC_HANDLE_PROMQL_ELAPSED: Histogram = register_histogram!(
         "frontend_handle_promql_elapsed",
```
Comment (evenyag) [inline src/frontend/src/metrics.rs:22]: missing prefix

**Comment by evenyag** [comment]
Adding a prefix is quite common and straightforward. Many projects use this way. For example, OpenDAL names its metrics like `opendal_request_duration_bucket`, not `request_duration_bucket`.

**Comment by zyy17** [comment]
> I am not sure if this is necessary, because we can always specify a database to write metrics from GreptimeDB that achieves the same isolation. Another choice is adding a special tag (for example, the cluster name or id) to all the metrics.
> 
> What do you think? @sunng87 @fengjiachun

Maybe it's another question. Adding a metric prefix is more like a naming convention to identify those metrics are from **greptimedb** and not easy to conflict with other metrics with the same same but from different services. If you operate multiple clusters, you have to use tags to clarify the same greptimedb metrics from different greptimedb and those tags can be added by other admin components (for example, greptimedb-operator).

**Comment by killme2008** [comment]
> > I am not sure if this is necessary, because we can always specify a database to write metrics from GreptimeDB that achieves the same isolation. Another choice is adding a special tag (for example, the cluster name or id) to all the metrics.
> > What do you think? @sunng87 @fengjiachun
> 
> Maybe it's another question. Adding a metric prefix is more like a naming convention to identify those metrics are from **greptimedb** and not easy to conflict with other metrics with the same same but from different services. If you operate multiple clusters, you have to use tags to clarify the same greptimedb metrics from different greptimedb and those tags can be added by other admin components (for example, greptimedb-operator).

Ok. Let's prefix them all, but I belive `greptime_` is better and shorter.

**Comment by sunng87** [comment]
We used to have @paomian's https://github.com/GreptimeTeam/greptimedb/pull/1590 for prefixing all metrics with `greptime`. But it losts during refactoring?

I'm +1 for prefixing our own metrics and leaving third-party's as it is.

**Comment by paomian** [comment]
We were using the layer function of the metrics library to add prefixes, then we switched to a different metrics library and there was no corresponding function. I think manually adding prefixes to our own metrics would be a good way to do it.

**Review by sunng87 — APPROVED** [review]

**Comment by killme2008** [comment]
I believe that we already added more metrics in recent PRs, you have to rebase with the main branch and ensure all metrics are renamed.

**Comment by zyy17** [comment]
@killme2008 I have already use `greptime_` as prefix and rebase the main branch.

**Review by killme2008 — APPROVED** [review]
LGTM

**Comment by killme2008** [comment]
@zhongzc I am not sure this PR has covered the metrics in the reverted index module. Let's merge it first, could you check the metrics name in the reverted index module?

**Review by zhongzc — APPROVED** [review]

**Comment by zhongzc** [comment]
> this PR has covered the metrics in the inverted index module

checked

**Pull request merged**